src/vs/platform/terminal/node/terminalEnvironment.ts

```diff
@@ -186,6 +187,8 @@ export function getShellIntegrationInjection(
 			// Move .zshrc into $ZDOTDIR as the way to activate the script
 			const zdotdir = path.join(os.tmpdir(), `${os.userInfo().username}-vscode-zsh`);
 			envMixin['ZDOTDIR'] = zdotdir;
+			const userZdotdir = env?.ZDOTDIR ?? os.homedir();
+			envMixin['USER_ZDOTDIR'] = userZdotdir;
```

Comment:
A comment explaining why this is needed would be good

Reply:
same explanation as above